translate/filters/test_checks.py

```diff
@@ -1365,23 +1365,160 @@ def test_skip_checks_per_language_in_some_checkers():
     """Test some checks are skipped for some languages in Mozilla checker."""
     from translate.storage import base
 
+    # Hijack checker config language ignoretests to test check is skipped.
+    checker_config = checks.CheckerConfig(targetlanguage="gl")
+    previous_ignoretests = checker_config.lang.ignoretests
+    checker_config.lang.ignoretests = {
+        'mozilla': ['accelerators'],
+    }
+
+    # Prepare the checkers and the unit.
+    mozillachecker = checks.MozillaChecker(checkerconfig=checker_config)
+    stdchecker = checks.StandardChecker(
+        checkerconfig=checks.CheckerConfig(accelmarkers="&",
+                                           targetlanguage="as")
+    )
+
     str1, str2, __ = strprep(u"&Check for updates", u"আপডেইটসমূহৰ বাবে নিৰীক্ষণ কৰক")
     unit = base.TranslationUnit(str1)
     unit.target = str2
 
+    # Accelerators check is disabled for Assamese in MozillaChecker.
+    assert 'accelerators' not in mozillachecker.run_filters(unit).keys()
+
+    # But it is not in StandardChecker.
+    assert 'accelerators' in stdchecker.run_filters(unit).keys()
+
+    # Undo hijack.
+    checker_config.lang.ignoretests = previous_ignoretests
+
+
+def test_mozilla_no_accelerators_for_indic():
+    """Test accelerators in MozillaChecker fails if accelerator in target.
+
+    No-accelerators is a special behavior of accelerators check in some
+    languages that is present in MozillaChecker.
+    """
     mozillachecker = checks.MozillaChecker(
         checkerconfig=checks.CheckerConfig(targetlanguage="as")
     )
-    failures = mozillachecker.run_filters(unit)
-    # Accelerators check is disabled for Assamese in MozillaChecker.
-    assert 'accelerators' not in failures.keys()
+    assert fails(mozillachecker.accelerators, "&File", "&Fayile")
+    assert fails(mozillachecker.accelerators, "My add&-ons", "&Byvoengs mit")
+    assert passes(mozillachecker.accelerators, "&File", "Fayile")
+    assert fails(mozillachecker.accelerators, "File", "&Fayile")
+    assert passes(mozillachecker.accelerators, "Mail &amp; News", "Po en Nuus")
+    assert fails(mozillachecker.accelerators, "Mail &amp; News", "Po en &Nuus")
+    assert passes(mozillachecker.accelerators, "&File", "Fayile")
```

Comment:
I suggest adding a sinple unlaced & as in "Mail & News" -> "Pos & Nuus", hopefully it works